shimmingtoolbox/load_nifti.py

```diff
@@ -111,14 +120,14 @@ def load_nifti(path_data, modality = 'phase'):
                 nifti_pos += 1
     else:
         niftis = np.empty([
-            info_init[0][0].shape[0], info_init[0][0].shape[1], info_init[0][0].shape[2],
-            echo_shape, info_init[0][0].shape[3]], dtype=float)
+            nifti_init[0][0].shape[0], nifti_init[0][0].shape[1], nifti_init[0][0].shape[2],
+            echo_shape, nifti_init[0][0].shape[3]], dtype=float)
         for i_echo in range(len(run_list[select_run])):
             if modality in run_list[select_run][i_echo][1]:
                 tmp_nii = run_list[select_run][i_echo][0]
                 info.append(tmp_nii[0].header)
                 json_info.append(tmp_nii[1])
-                for i_volume in range(info_init[0][0].shape[3]):
+                for i_volume in range(nifti_init[0][0].shape[3]):
                     niftis[:, :, :, nifti_pos, i_volume] = tmp_nii[2][:, :, :, i_volume]
```

Comment:
Other than this inner for-loop, the enclosing for loop is a duplicate of line 114-120. This is out of scope for the current PR, but is a prime candidate for refactoring in the future. The duplicated code could be factored out into a function taking a boolean param that only runs the inner for-loop for the appropriate case. Also, depending on the size of `range(len(run_list[select_run]))` and `range(nifti_init[0][0].shape[3])`, the `map` function could be used to achieve better performance.

Reply:
It's true that I doubled my code. I will try to remove it. I don't think a separate function will be usefull since it will be used only these two times and that we can merge the duplicated code into a more condensed code.